### DISCUSSION TITLE
Parent/child interfaces between devices and VMs

### DISCUSSION THREAD

**Discussion by empusas** [netbox-community/netbox#8327] [Q&A]
So we can add devices to a cluster and assign virtual machines to it.
But when I create an VM interface it cannot have a parent interface on the physical device that the cluster consist of.

How is this intended to be used?

Would it not beneficial if I can map the VMs virtual interface to the physical parent interface on the device?
| Parent   |      Child      | 
|----------|-------------|
Device-eth0 | VM-eth0.34

**Comment by candlerb** [comment]
"Parent" is used to relate virtual interfaces or sub-interfaces on a device to a real interface on the same device: e.g. `Gi0/1.100` is a sub-interface of `Gi0/1` (and hence traffic for `Gi0/1.100` always goes in and out of `Gi0/1`).  Another use case is where an IPMI interface shares the same port as a data port.

There's no direct way to associate a given VM's interface to a physical device interface, because the model assumes that a VM can be running on a cluster, and the VM may migrate between nodes on a cluster.  In other words, there isn't even a direct relationship between VM and a single Device (unless the cluster happens to consist of a single device, but in general that may not be the case).

What I suggest is that you define the VM interface as 802.1Q mode "access" with native VLAN ID 34 - i.e. first create a VLAN object with VID 34.  This will directly model the broadcast domain that the VM is connected to.  Also model the physical device eth0 as being a member of VLAN 34 (which might be access or trunk).  This then relates both the VM and the device port to the same broadcast domain.

**Reply by ovizii** [reply]
Thanks for this. I've been trying to figure this out the whole day :-)

**Comment by jdavidson2021** [comment]
Is there not a way to connect a VM interface to a bridge on a specific server?  Assuming that you created a bridge interface on the host server, you would think that this host server bridge interfaces would be passed into the list of bridge interface or parent interfaces on the VM so that you can tie a VM virtual interface to a parent interface on the host server.
